docs/sources/style-guide/write-for-developers/index.md

```diff
@@ -0,0 +1,153 @@
+
+When providing an example, give a written description. You can put it either in the body of the document or as explanatory comments within the example code.
+
+Remember the basic rule: explain _why_ your code does what it does, rather than describe _what_ it does.
+For an in-depth, external resource about writing developer documentation, see [_Docs for Developers_](https://docsfordevelopers.com/).
+
+### Formatting code examples
+
+Here are guidelines to follow when formatting code examples.
+
+- Use spaces, not tabs. 
+- Follow Grafana's accepted [coding style guidelines](https://github.com/grafana/grafana/blob/main/contribute/style-guides/).
+- Wrap lines at 80 characters.
+- When omitting code, use three dots (...). Don't use the ellipsis character (…).
+
+Introduce each code sample with a sentence or paragraph to establish its context. End the introduction with a colon if it immediately precedes the sample, or a period if it doesn't. 
+
+### Paths, filenames, and URLs
+
+Many types of information belong in fixed-width font. Among these are path names, file names, directories, or folders. However, don't format domain names or URLs as code if you intend the user to follow the link.
+
+Although there are times when you might want to make a URL clickable, it is better to follow the guidelines for [References]({{< relref "../../writing-guide/references/" >}}).
+
+## Command lines
+
+Use the following conventions when you include commands on the command line in technical content.
+
+- Do not assume everyone is using Linux. Make sure instructions include enough information for Windows and Mac users to successfully complete procedures.
+
+- Do not add `$` before commands. Make it easy for users to copy and paste commands.
+
+  - **Right:** `sudo yum install grafana`
+  - **Wrong:** `$ sudo yum install grafana`
+
+- Include `sudo` before commands that require `sudo` to work.
+
+For terminal examples and configurations, use a `bash` code block:
+
+```bash
+sudo yum install grafana
+```
+
+If your command-line instructions include a combination of input and output lines, use separate code blocks for input and output, and use a `console` code block for the output.
+
+```bash
+cat ~/.ssh/my-ssh-key.pub
+```
+
+The output is similar to the following:
+
+```console
+ssh-rsa KEY_VALUE USERNAME
+```
+
+For HTTP request/response, use an `http` code block:
+
+```http
+GET /api/dashboards/id/1/permissions HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+```
+
+If an argument is optional, enclose it in square brackets.
+
+```console
+ssh-rsa KEY_VALUE USERNAME [_FILENAME_]
+```
+
+Use descriptive words and phrases when including placeholders, and avoid using X or XXX.
+In Markdown, in front of a placeholder, use an asterisk followed by a backtick. At the end of the placeholder, use a backtick followed by an asterisk.
+
+ For example: *`A_PLACEHOLDER`*.
```

Comment:
i feel inclined to suggest, for this case and others, that the examples should be shown in both their literal raw input and rendered markdown form. Here, that might mean something like this instead:

---

...use a backtick followed by an asterisk. For example, prefer the following raw markdown:

```
The following text is a placeholder: *`A_PLACEHOLDER`*.
```
to produce:

> The following text is a placeholder: *`A_PLACEHOLDER`*.

---

Alternatively, a table might be used showing raw inputs to more tidily separate raw inputs and rendered markdown outputs.

Otherwise, someone reading this document in rendered markdown will have to flip between viewing markdown source and the rendered document in order to be absolutely sure of the input syntax being recommended.

https://pkg.go.dev/github.com/fluhus/godoc-tricks uses this approach to quite positive effect - it immediately organically became my go-to reference for writing Go docs with the new formatting capabilities introduced in 1.19.

Reply:
Perhaps we can use a table format in the future, but for now I modified the code examples using the pattern that you suggested. Please re-review.